xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -5232,20 +5245,12 @@ glusterd_import_friend_volumes_synctask(void *opaque)
     if (!arg)
         goto out;
 
-    peer_data = dict_new();
+    peer_data = arg->peer_data;
     if (!peer_data) {
         gf_smsg(this->name, GF_LOG_ERROR, errno, GD_MSG_DICT_CREATE_FAIL, NULL);
         goto out;
     }
```

Comment:
I think this is not needed. If `arg->peer_data` is not valid, we would have failed much before getting here.